engine/access/rpc/backend/backend_transactions.go

```diff
@@ -240,20 +240,67 @@ func (b *backendTransactions) GetTransactionResult(
 	}
 
 	// derive status of the transaction
-	status, err := b.deriveTransactionStatus(tx, transactionWasExecuted, block)
+	txStatus, err := b.deriveTransactionStatus(tx, transactionWasExecuted, block)
 	if err != nil {
 		return nil, convertStorageError(err)
 	}
 
 	return &access.TransactionResult{
-		Status:       status,
+		Status:       txStatus,
 		StatusCode:   uint(statusCode),
 		Events:       events,
 		ErrorMessage: txError,
 		BlockID:      blockID,
 	}, nil
 }
 
+func (b *backendTransactions) GetTransactionResultByIndex(
+	ctx context.Context,
+	blockID flow.Identifier,
+	index uint32,
+) (*access.TransactionResult, error) {
+	// TODO: Merge flow.TransactionResult and access.TransactionResult so caching doesn't cause a circular dependency
+	// create request and forward to EN
+	req := execproto.GetTransactionByIndexRequest{
+		BlockId: blockID[:],
+		Index:   index,
+	}
+	execNodes, err := executionNodesForBlockID(ctx, blockID, b.executionReceipts, b.state, b.log)
+	if err != nil {
+		if errors.As(err, &InsufficientExecutionReceipts{}) {
+			return nil, status.Errorf(codes.NotFound, err.Error())
+		}
+		return nil, status.Errorf(codes.Internal, "failed to retrieve result from any execution node: %v", err)
+	}
+
+	resp, err := b.getTransactionResultByIndexFromAnyExeNode(ctx, execNodes, req)
+	if err != nil {
+		if status.Code(err) == codes.NotFound {
+			return nil, err
+		}
+		return nil, status.Errorf(codes.Internal, "failed to retrieve result from execution node: %v", err)
+	}
+
+	block, err := b.blocks.ByID(blockID)
```

Comment:
Hmm, maybe we can query blocks before we issue request to EN? I mean if block isn't there we created unnecessary load